packages/webapp/src/plugins/primary-page.js

```diff
@@ -0,0 +1,20 @@
+import constants from '../utils/constants.js'
+
+const primaryPage = {
+  name: 'primary-page',
+  register: (server, _options) => {
+    const routeMapping = constants.primaryPages || {}
+
+    server.ext('onPostHandler', (request, h) => {
+      const applicationType = request.yar.get(constants.redisKeys.APPLICATION_TYPE)
+      const targetRoutes = routeMapping[applicationType] || []
+
+      if (targetRoutes.includes(request.path)) {
+        request.yar.set(constants.redisKeys.PRIMARY_ROUTE, request.path)
```

Comment:
Set the primary route - tasklist or check and submit page for example when the user goes to it. Then we can take them back there once tasks are completed (based on the nextUrl of a route definition returning the primary route value).